src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -1758,11 +1758,12 @@ namespace HeatBalanceManager {
                 // Find the material in the list of materials
 
                 thisConstruct.LayerPoint(Layer) = UtilityRoutines::FindItemInPtrList(ConstructAlphas(Layer), state.dataMaterial->Material);
-
                 // count number of glass layers
                 if (thisConstruct.LayerPoint(Layer) > 0) {
-                    if (state.dataMaterial->Material(thisConstruct.LayerPoint(Layer))->Group == Material::MaterialGroup::WindowGlass) ++iMatGlass;
-                    MaterialLayerGroup = state.dataMaterial->Material(thisConstruct.LayerPoint(Layer))->Group;
+                    auto const *thisMaterial = state.dataMaterial->Material(thisConstruct.LayerPoint(Layer));
+                    auto const *thisMaterialChild = dynamic_cast<const Material::MaterialChild *>(thisMaterial);
```

Comment:
If you only need the `Group` field, you don't have to do the `dynamic cast`. That should accessible from `MaterialBase`.